### ISSUE TITLE
W0613 erroneous for abc.abstractclassmethod

### ISSUE THREAD

**Issue by lhfriedman** [PyCQA/pylint#3827]
```py
import abc

class A(abc.ABC):
    @abc.abstractmethod
    def meth_0(self, x):
        ...

class B(abc.ABC):
    @abc.abstractclassmethod
    def meth_0(cls, x):
        ...

```
Running pylint produces an erroneous warning W0613 for line 10. This should probably be warning free, the same as line 5. The
only difference is class vs instance method.
```

> pylint.bat abstractmethoderror.py                                                         
************* Module abstractmethoderror
playground\abstractmethoderror.py:1:0: C0114: Missing module docstring (missing-module-docstring)
playground\abstractmethoderror.py:3:0: C0103: Class name "A" doesn't conform to PascalCase naming style (invalid-name)
playground\abstractmethoderror.py:3:0: C0115: Missing class docstring (missing-class-docstring)
playground\abstractmethoderror.py:5:4: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
playground\abstractmethoderror.py:5:4: C0116: Missing function or method docstring (missing-function-docstring)
playground\abstractmethoderror.py:3:0: R0903: Too few public methods (1/2) (too-few-public-methods)
playground\abstractmethoderror.py:8:0: C0103: Class name "B" doesn't conform to PascalCase naming style (invalid-name)
playground\abstractmethoderror.py:8:0: C0115: Missing class docstring (missing-class-docstring)
playground\abstractmethoderror.py:10:4: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
playground\abstractmethoderror.py:10:4: C0116: Missing function or method docstring (missing-function-docstring)
playground\abstractmethoderror.py:10:20: W0613: Unused argument 'x' (unused-argument)
playground\abstractmethoderror.py:8:0: R0903: Too few public methods (1/2) (too-few-public-methods)

--------------------------------------------------------------------
Your code has been rated at -7.14/10 (previous run: -5.71/10, -1.43)
```
```
> pylint --version                                                                                       
pylint 2.5.3
astroid 2.4.2
Python 3.7.7 (default, May  6 2020, 11:45:54) [MSC v.1916 64 bit (AMD64)]

```

**Comment by AWhetter** [comment]
What's in `meth_0` of both methods? Are they using `x`?

**Comment by Pierre-Sassoulas** [comment]
Same remark, the most important part of the code is left out in the example.

With:
```
import abc

class A(abc.ABC):
    @abc.abstractmethod
    def meth_0(self, x):
        pass

class B(abc.ABC):
    @abc.abstractclassmethod
    def meth_0(cls, x):
        pass
```

The result is: 
```
taest.py:1:0: C0114: Missing module docstring (missing-module-docstring)
taest.py:3:0: C0103: Class name "A" doesn't conform to PascalCase naming style (invalid-name)
taest.py:3:0: C0115: Missing class docstring (missing-class-docstring)
taest.py:5:4: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
taest.py:5:4: C0116: Missing function or method docstring (missing-function-docstring)
taest.py:3:0: R0903: Too few public methods (1/2) (too-few-public-methods)
taest.py:8:0: C0103: Class name "B" doesn't conform to PascalCase naming style (invalid-name)
taest.py:8:0: C0115: Missing class docstring (missing-class-docstring)
taest.py:10:4: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)
taest.py:10:4: C0116: Missing function or method docstring (missing-function-docstring)
taest.py:8:0: R0903: Too few public methods (1/2) (too-few-public-methods)
```

**Comment by lhfriedman** [comment]
> What's in `meth_0` of both methods? Are they using `x`?

@AWhetter , _There is no part of the code missing._ 

Ellipsis `...` is in place of `pass` statement. As I understand it, this is acceptable / preferred code to using `pass` for abstract methods in Python 3. It also appears to be what is generating the error, so thanks for pointing that out, @Pierre-Sassoulas.

Notice that pylint is happy with class A, but not class B in the example code. Shouldn't they equally prefer or disallow the use of ellipsis? Is ellipsis really OK for abstract methods but not abstract **class** methods?

**Comment by lhfriedman** [comment]
Also, please remove the `cannot reproduce` label. I think I addressed the ellipsis issue adequately. See from original post:

> Running pylint produces an erroneous warning W0613 for line 10. This should probably be warning free, the same as line 5. The only difference is class vs instance method.

**Comment by Pierre-Sassoulas** [comment]
Closing as ``@abc.abstractclassmethod`` is deprecated since python 3.3 and replacing with ``@abc.abstractmethod`` + ``@classmethod`` does not raise the false positive.
